docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+The hypervisor uses an image file which provides a minimal root
+filesystem used by the guest kernel to boot the VM and host the Kata Container.
+
+Kata Containers supports both initrd and rootfs based minimal guest images.
+
+The default images are created using the [`osbuilder`](../../tools/osbuilder) tool.
+
+> **Note:**
+>
+> The guest image is *unrelated* to the image used in a container
+> workload.
+>
+> For example, if a user creates a container that runs a shell in a
+> BusyBox image, they will run that shell in a BusyBox environment.
+> However, the guest image running inside the VM that is used to
+> *host* that BusyBox image could be running Clear Linux, Ubuntu,
+> Fedora or any other distribution potentially.
+>
+> The `osbuilder` tool provides
+> [configurations for various common Linux distributions](../../tools/osbuilder/rootfs-builder)
+> which can be built into either initrd or rootfs guest images.
 
 #### Root filesystem image
 
-The default packaged root filesystem image, sometimes referred to as the "mini O/S", is a
-highly optimized container bootstrap system based on [Clear Linux](https://clearlinux.org/). It provides an extremely minimal environment and
-has a highly optimized boot path.
+The default packaged root filesystem image ("rootfs"), sometimes
+referred to as the "mini O/S", is a highly optimized container
+bootstrap system based on [Clear Linux](https://clearlinux.org). It
+provides an extremely minimal environment and has a highly optimized
+boot path.
+
+The only services running in the context of the mini O/S are:
+
+| Process | Environment | systemd service? | Notes |
+|-|-|-|-|
+| systemd | VM root | n/a | The init daemon, running as PID 1 |
+| [Agent](#agent) | VM root | yes | Runs as a systemd service |
+| `chronyd` | VM root | yes | Used to synchronise the time with the host |
+| container workload | VM container | no | Managed by the agent |
+
+See also the [process overview](#process-overview).
 
-The only services running in the context of the mini O/S are the init daemon
-(`systemd`) and the [Agent](#agent). The real workload the user wishes to run
-is created using libcontainer, creating a container in the same manner that is done
-by `runc`.
+> **Note:** The container workload is running inside a full container
+> environment which itself is running within a VM environment.
 
-For example, when `ctr run -ti ubuntu date` is run:
+For example, when the user runs `ctr run -ti ubuntu date`:
 
 - The hypervisor will boot the mini-OS image using the guest kernel.
 - `systemd`, running inside the mini-OS context, will launch the `kata-agent` in
   the same context.
 - The agent will create a new confined context to run the specified command in
   (`date` in this example).
 - The agent will then execute the command (`date` in this example) inside this
-  new context, first setting the root filesystem to the expected Ubuntu\* root
+  new context, first setting the root filesystem to the expected Ubuntu root
```

Comment:
This line left me confused.  It sounds like Ubuntu is the rootfs but isn't Ubuntu the container image?  It seems that rootfs is being used interchangeably for the VM rootfs and the containers file system which I don't know if that is actually a rootfs.

Reply:
I am still confused :)   For the container workload are you referring a container image like Nginx that has the Nginx application inside of a containerized image based on Ubuntu or another OS?  That is what I think of when I think container workload. However, under user accessible you put "exec command" with a link that when you click it says  "The exec command allows an administrator or developer to enter the VM environment which is inaccessible to the container workload."   I didn't think the container workload would have any access to the VM.

Reply:
This is all explained in the "Environments" section with concrete examples of the terms. I've clarified the exec command text a little but the key word in that sentence is "**in**accessible". Did you possibly mis-read that as "accessible"? I've reworded this to now say:

> The exec command allows an administrator or developer to enter the VM root environment which is not accessible by the container workload.

Reply:
No I read it as the "kubectl" exec command allows a developer into the VM environment but the container workload doesn't have access to that environment.  I was thinking of exec as in  "kubectl exec -it ..."  and maybe that is why I am confused.  I did not think anyone with "kubectl exec" access would have access to the VM environment. I think exec in this example is being used in a different way and isn't kubernetes related.